pkg/docker/docker.go

```diff
@@ -46,9 +48,14 @@ func (d *Docker) DoesServiceExist(ctx context.Context, name string) bool {
 	if err != nil || len(list) == 0 {
 		return false
 	}
+	suffix := fmt.Sprintf("_%s", name)
 	for _, service := range list {
-		if service.Spec.Annotations.Name == name {
-			return true
+		if strings.HasSuffix(service.Spec.Annotations.Name, suffix) {
+			for _, name := range service.Spec.Labels {
+				if name == "infrastructure" {
```

Comment:
I recommend closing this PR for now, but you can decide if you want to update it or not.